pkg/endpoint/endpoint.go

```diff
@@ -2387,6 +2387,15 @@ func (e *Endpoint) syncPolicyMap() error {
 		e.realizedMapState = make(map[policymap.PolicyKey]struct{})
 	}
 
+	if e.desiredMapState == nil {
+		e.desiredMapState = make(map[policymap.PolicyKey]struct{})
+	}
+
+	if e.PolicyMap == nil {
+		e.getLogger().Debug("not syncing PolicyMap state for endpoint because PolicyMap is nil")
+		return nil
```

Comment:
I agree, let's return an error here so we retry before the regular interval as we will likely recover soon from the build error.